idp/authmiddleware_test.go

```diff
@@ -663,6 +788,24 @@ func (spy *handlerSpy) assertPrincipalIs(expectedPrincipal scim.Principal) error
 	return nil
 }
 
+func (spy *handlerSpy) assertExternalPrincipalIs(expectedExternalPrincipal scim.Principal) error {
```

Comment:
On the phone we discussed that it would be useful to have a separate function for the check. I have now removed it again and switched to the existing function.